pkg/registry/apis/alerting/notifications/timeinterval/conversions.go

```diff
@@ -30,23 +31,15 @@ func convertToK8sResources(orgID int64, intervals []definitions.MuteTimeInterval
 		return nil, err
 	}
 	result := &model.TimeIntervalList{}
+
 	for idx := range specs {
 		interval := intervals[idx]
 		spec := specs[idx]
-		uid := getIntervalUID(interval) // TODO replace to stable UID when we switch to normal storage
-		result.Items = append(result.Items, model.TimeInterval{
-			TypeMeta: resourceInfo.TypeMeta(),
-			ObjectMeta: metav1.ObjectMeta{
-				UID:       types.UID(uid), // TODO This is needed to make PATCH work
-				Name:      uid,            // TODO replace to stable UID when we switch to normal storage
-				Namespace: namespacer(orgID),
-				Annotations: map[string]string{ // TODO find a better place for provenance?
-					"grafana.com/provenance": string(interval.Provenance),
-				},
-				ResourceVersion: interval.Version,
-			},
-			Spec: spec,
-		})
+		item := buildTimeInterval(orgID, interval, spec, namespacer)
+		if selector != nil && !selector.Empty() && !selector.Matches(model.SelectableTimeIntervalsFields(&item)) {
```

Comment:
NIT: Is `!selector.Empty()` necessary?

Reply:
The idea is to create field set only when it's really necessary. The selector is an interface, so implementation may return true in some cases.